### PR TITLE
Bucket lists

### PR DESCRIPTION
**Pull request by evelynhasama**
User can create bucket lists (with photo, name, description, and friends), view their bucket lists, and sort/filter their lists.

### CODE REVIEW

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java:53]: nit: I'm scratching my head around these `TYPE` things, can you please add a comment with a tl;dr?

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java:74]: nit: Code dedup opportunity
```
String url = user.getImage() != null ? user.getImage().getUrl() : R.drawable.profile_placeholder;
Glide.with(context.load(url).circleCrop().into(friendViewHolder.ivProfileImage);
```

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java:81]: nit: Code dedupe opportunity.

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java:114]: nit: Code dedup opportunity.

File: app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:102]: <3

Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:102]: nit: why `friendsSave` name?

File: app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:164]: very nice:

File: app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:212]: nit: Consider flipping the `if` and early returning to reduce nesting:
```
if(data == null || requestCode != PICK_PHOTO_CODE) {
  // nothing to do
  return;
}
....
```

File: app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:232]: nit: why a `switch` if there's only one condition?

File: app/src/main/java/com/evelynhasama/bucketbuds/FriendsAdapter.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/FriendsAdapter.java:169]: nit: Why `Log.d` instead of `Log.e`?
Is the exception expected in some scenarios?

IF the exception is truly an exception (you don't expect it to happen), then `Log.e` is correct.

If there are _some_ exceptions that are expected, then you can use different `Log.e/Log.w/Log.i` depending on what you expect or not.

File: app/src/main/java/com/evelynhasama/bucketbuds/FriendsPageFragment.java
Comment (SantiCe) [inline app/src/main/java/com/evelynhasama/bucketbuds/FriendsPageFragment.java:141]: Why not use the `User` constants if you are querying `Users`?

**Review by SantiCe — CHANGES_REQUESTED** [review]
Feature seems to work nicely! 
Requesting some style changes but overall looks good.

**Comment by SantiCe** [comment]
Feel free to merge after addressing the comments, no need to wait for approval unless there's something specific that you want me to look at after.

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java
Comment (evelynhasama) [reply app/src/main/java/com/evelynhasama/bucketbuds/BucketFriendsAdapter.java:74]: R.drawable.profile_placeholder is an integer, so they cannot be put in the same variable

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java
Comment (evelynhasama) [reply app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java:114]: yeah I thought about this one when I was writing it, but wasn't sure how to implement since one is ascendingOrder and the other two are descendingOrder. any ideas?

File: app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java
Comment (evelynhasama) [reply app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java:232]: yeah that was unnecessary😔

File: app/src/main/java/com/evelynhasama/bucketbuds/FriendsAdapter.java
Comment (evelynhasama) [reply app/src/main/java/com/evelynhasama/bucketbuds/FriendsAdapter.java:169]: I got really used to using Log.d for debugging and decided to use them for everything, but yeah I should use the appropriate Log.

File: app/src/main/java/com/evelynhasama/bucketbuds/FriendsPageFragment.java
Comment (evelynhasama) [reply app/src/main/java/com/evelynhasama/bucketbuds/FriendsPageFragment.java:141]: good point!

File: app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java
Comment (SantiCe) [reply app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java:114]:
```
ParseQuery query = userPub.getBucketsRelation().getQuery();
switch(..) {
  case R.id.rbAlphabetical:
    query = query.addAscendingOrder(BucketList.KEY_NAME);
.....
}
query.findInBackground(bucketListFindCallback);
```

**Review by SantiCe — APPROVED** [review]

**Pull request merged**
